OPHD/Map/Tile.h

```diff
@@ -15,7 +15,8 @@
 class Tile
 {
 public:
-	Tile();
+	Tile() = default;
+	Tile(Tile&& other);
```

Comment:
Rule of Six violation. Either default, explicitly define or delete a copy ctor for shallow copies, deep copies, or making `Tile` non-copyable respectively.

Right now, I would recommend a deep copy since the tile owns `mMine` but not `mThing`...which makes no sense. (See next).

It shouldn't own either (in that case, a `default` would work). There should be an owned-by-map, map-wide, collection of entities (`std::map<Point<int>,std::unique_ptr<Thing>> mThings;`) and another (optional, albeit makes referencing *just* structures easier with a range-based for) non-owning collection of `Structures*`: `std::vector<Structure*>`.

Reply:
The destructor tries to delete both pointers. That suggests ownership of both those pointers.

I think copy should be disallowed. A move only class would be easier to deal with, and I don't think allowing copies is intentional.